Composer/packages/client/src/recoilModel/dispatchers/trigger.ts

```diff
@@ -0,0 +1,179 @@
+          const designerId1 = getDesignerIdFromDialogPath(
+            newDialog,
+            `content.triggers[${index}].actions[0].actions[1].prompt`
+          );
+          const designerId2 = getDesignerIdFromDialogPath(
+            newDialog,
+            `content.triggers[${index}].actions[0].elseActions[0].activity`
+          );
+          const lgTemplates: LgTemplate[] = [
+            LgTemplateSamples.TextInputPromptForQnAMatcher(designerId1) as LgTemplate,
+            LgTemplateSamples.SendActivityForQnAMatcher(designerId2) as LgTemplate,
+          ];
+          await createLgTemplates({ id: lgFile.id, templates: lgTemplates, projectId });
+        } else if (formData.$kind === onChooseIntentKey) {
+          const designerId1 = getDesignerIdFromDialogPath(newDialog, `content.triggers[${index}].actions[4].prompt`);
+          const designerId2 = getDesignerIdFromDialogPath(
+            newDialog,
+            `content.triggers[${index}].actions[5].elseActions[0].activity`
+          );
+          const lgTemplates1: LgTemplate[] = [
+            LgTemplateSamples.TextInputPromptForOnChooseIntent(designerId1) as LgTemplate,
+            LgTemplateSamples.SendActivityForOnChooseIntent(designerId2) as LgTemplate,
+          ];
+
+          let lgTemplates2: LgTemplate[] = [
+            LgTemplateSamples.adaptiveCardJson as LgTemplate,
+            LgTemplateSamples.whichOneDidYouMean as LgTemplate,
+            LgTemplateSamples.pickOne as LgTemplate,
+            LgTemplateSamples.getAnswerReadBack as LgTemplate,
+            LgTemplateSamples.getIntentReadBack as LgTemplate,
+          ];
+          const commonlgFile = lgFiles.find(({ id }) => id === `common.${locale}`);
+
+          lgTemplates2 = lgTemplates2.filter(
+            (t) => commonlgFile?.templates.findIndex((clft) => clft.name === t.name) === -1
+          );
+
+          await createLgTemplates({ id: `common.${locale}`, templates: lgTemplates2, projectId });
+          await createLgTemplates({ id: lgFile.id, templates: lgTemplates1, projectId });
+        }
+        const dialogPayload = {
+          id: newDialog.id,
+          projectId,
+          content: newDialog.content,
+        };
+        await updateDialog(dialogPayload);
+        if (autoSelected) {
+          selectTo(projectId, newDialog.id, `triggers[${index}]`);
+        }
+      } catch (ex) {
+        setError(callbackHelpers, ex);
+      }
+    }
+  );
+
+  const deleteTrigger = useRecoilCallback(
+    (callbackHelpers: CallbackInterface) => async (projectId: string, dialogId: string, trigger: ITriggerCondition) => {
+      try {
+        const { snapshot } = callbackHelpers;
+        const dispatcher = await snapshot.getPromise(dispatcherState);
+
+        const { removeLuIntent, removeLgTemplates } = dispatcher;
+
+        if (get(trigger, '$kind') === SDKKinds.OnIntent) {
+          const intentName = get(trigger, 'intent', '') as string;
```

Comment:
is casting to as string required?

Reply:
Sadly, I think it is. `get` ruins type detection, so if you want to keep TypeScript happy you need to cast stuff.

Reply:
I updated it to directly use `trigger.$kind`